src/libs/ajax/User.test.ts

```diff
@@ -1,38 +1,441 @@
-import { kvArrayToObject } from './User';
-
-// Workaround for circular import issues.
-jest.mock('src/auth/auth');
-
-describe('kvArrayToObject', () => {
-  it('converts an array of key/value objects to an object', () => {
-    // Act
-    const result = kvArrayToObject([
-      { key: 'foo', value: 1 },
-      { key: 'bar', value: 2 },
-      { key: 'baz', value: 3 },
-    ]);
-
-    // Assert
-    expect(result).toEqual({
-      foo: 1,
-      bar: 2,
-      baz: 3,
+import {
+  CreateTerraUserProfileRequest,
+  generateAPIBodyForCreateUserProfile,
+  generateAPIBodyForUpdateUserProfile,
+  OrchestrationUpsertTerraUserProfileRequest,
+  UpdateTerraUserProfileRequest,
+} from 'src/libs/ajax/User';
+import { TerraUserProfile } from 'src/libs/state';
+
+jest.mock('src/libs/ajax/ajax-common', () => ({
+  authOpts: jest.fn(() => ''),
+}));
+
+const completeUserProfile: TerraUserProfile = {
+  firstName: 'testFirstName',
+  lastName: 'testFirstName',
+  contactEmail: 'testFirstName',
+  department: 'testDepartment',
+  institute: 'testInstitute',
+  interestInTerra: 'testInterestInTerra',
+  programLocationCity: 'testCity',
+  programLocationCountry: 'testCountry',
+  programLocationState: 'testState',
+  starredWorkspaces: 'testStarredWorkspaces',
+  title: 'testTitle',
+  researchArea: 'testResearchArea',
+};
+
+const minimalCreateUserProfileRequest: CreateTerraUserProfileRequest = {
+  firstName: completeUserProfile.firstName!,
```

Comment:
Many of these non-null assertions could be avoided by typing `completeUserProfile` as a const.

```ts
const completeUserProfile = {
  ...
} as const satisfies TerraUserProfile;
```